### PR TITLE
[FIX] don't expect a singleton when searching for

### PR DESCRIPTION
**Pull request by hbrunn**
recipient objects

### CODE REVIEW

File: newsletter/model/newsletter_newsletter.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def action_preview(self):
                 'template_id': self.type_id.email_template_id.id,
                 'default_res_id': self.id,
                 'newsletter_res_id': self.env[self.type_id.model.model].search(
-                    safe_eval(self.type_id.domain)).id or False,
+                    safe_eval(self.type_id.domain)).ids[-1] or False,
```
Comment (pedrobaeza) [inline newsletter/model/newsletter_newsletter.py]: It's better to make `safe_eval(self.type_id.domain))[:1].id` so you handle all cases (even empty recordset)

**Comment by hbrunn** [comment]
@pedrobaeza thanks, much better to use a slice here

**Comment by guewen** [comment]
:+1:

**Comment by oihane** [comment]
:+1:

File: newsletter/model/newsletter_newsletter.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def action_preview(self):
                 'template_id': self.type_id.email_template_id.id,
                 'default_res_id': self.id,
                 'newsletter_res_id': self.env[self.type_id.model.model].search(
-                    safe_eval(self.type_id.domain)).id or False,
+                    safe_eval(self.type_id.domain)).ids[:-1] or False,
```
Comment (pedrobaeza) [inline newsletter/model/newsletter_newsletter.py:71]: This is returning the last element, not the first, which can be semantically different (not sure about this case). That's why I told you to put `.ids[:1]`

Comment (hbrunn) [inline newsletter/model/newsletter_newsletter.py:71]: @pedrobaeza it's arbitrary. we just need to pick some record for the preview, and i figured the last object is nicer in some cases. But this totally depends on the sort order of said object and doesn't matter in the end

Comment (pedrobaeza) [inline newsletter/model/newsletter_newsletter.py:71]: :+1: then

**Pull request merged**